src/components/Molecules/BlockDetails/BlockDetails.tsx

```diff
@@ -0,0 +1,73 @@
+import { Card, CardDescription, CardTitle } from "@/components/ui/card";
+import { Skeleton } from "@/components/ui/skeleton";
+import { GRK_SIZES } from "@/utils/constants/shared.constants";
+import { timestampParser } from "@/utils/functions";
+import { None, Some, type Option } from "@/utils/option";
+import { useCovalent } from "@/utils/store/Covalent";
+import { type BlockDetailsProps } from "@/utils/types/molecules.types";
+import { type Block } from "@covalenthq/client-sdk";
+import { useEffect, useState } from "react";
+
+export const BlockDetails: React.FC<BlockDetailsProps> = ({
+    chain_name,
+    height,
+}) => {
+    const { covalentClient } = useCovalent();
+
+    const [maybeResult, setResult] = useState<Option<Block>>(None);
+
+    useEffect(() => {
+        (async () => {
+            setResult(None);
+            try {
+                const { data, ...error } =
+                    await covalentClient.BaseService.getBlock(
+                        chain_name,
+                        height.toString()
+                    );
+                if (error.error) {
```

Comment:
Instead of throwing an error can you display an error message within the component. The user might not know what is happening since it just shows the skeleton. Most of the other components will render the error message

Reply:
Fixed in: dea313e